sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/base_node.py

```diff
@@ -365,8 +365,11 @@ def _from_rest_object(cls, obj: dict) -> "BaseNode":
             obj[CommonYamlFields.TYPE] = NodeType.COMMAND
 
         from azure.ai.ml.entities._job.pipeline._load_component import pipeline_node_factory
-
-        instance: BaseNode = pipeline_node_factory.get_create_instance_func(obj[CommonYamlFields.TYPE])()
+        # todo: refine Hard code for now to support different task type for DataTransfer node
```

Comment:
mark

Reply:
what will we send to server-side?

Reply:
This is because we need register different component and node type for three tasks in datatransfer.

No more ideas on this for now, so add todo here

Reply:
you can refer to relevant tests case, e.g. ```test_pipeline_with_data_transfer_copy_function```